src/ServerHandler.cpp

```diff
@@ -1,8 +1,153 @@
 //
-// Created by idanto on 11/01/2020.
+// Created by idanto on 13/01/2020.
 //
 
 #include "../include/ServerHandler.h"
 
 
-short port = boost::lexical_cast<short>(cmdParams[1].substr(portSeparator, cmdParams[1].size()));
\ No newline at end of file
+ServerHandler::ServerHandler(ConcurrentDataQueues& queues, ClientInventory& inventory) :
+        _connectionHandler(nullptr), _queues(queues), _inventory(inventory) {}
+
+ServerHandler::~ServerHandler() {
+    delete _connectionHandler;
+}
+
+void ServerHandler::split(std::string& str, std::vector <std::string>& subStrs, std::string delimiter) {
+    size_t start = 0U;
+    size_t end = str.find(delimiter);
+    while (end != std::string::npos) {
+        subStrs.push_back(str.substr(start, end - start));
+        start = end + delimiter.length();
+        end = str.find(delimiter, start);
+    }
+    subStrs.push_back(str.substr(start, str.size() - start + 1));
+}
+
+Frame ServerHandler::receiveFrameFromClient() {
+    boost::unique_lock <boost::mutex> lock(_queues.mutexToServer);
+    Frame frame;
+    if (!_queues.framesToServer.empty()) {
+        frame = _queues.framesToServer.front();
+        _queues.framesToServer.pop();
+    }
+    return frame;
+}
+
+void ServerHandler::sendFrameToClient(Frame frame) {
+    boost::unique_lock <boost::mutex> lock(_queues.mutexFromServer);
+    _queues.framesFromServer.push(frame);
+    _queues.condFromServer.notify_all();
+}
+
+void ServerHandler::parseUserFrame(Frame frameFromClient) {
+    FrameCommand cmd = frameFromClient.getCommand();
+    switch (cmd) {
+        case CONNECT: {
+            std::string host = frameFromClient.getHeaderVal("host");
+            short port = boost::lexical_cast<short>(frameFromClient.getHeaderVal("port"));
+            _connectionHandler = new ConnectionHandler(host, port);
+            if (!_connectionHandler->connect()) {
+                Frame frame = Frame(ERROR, "Could not connect to server");
+            }
+            else {
+                _connectionHandler->sendFrameAscii(frameFromClient.toString(), STOMP_DELIMITER.c_str()[0]);
```

Comment:
Considering frameFromClient contains a port header, is it what we want sent to the server? Because this seems like it is the command the user sent. Without the required headers to make a STOMP CONNECT message (like version for example)

Reply:
see UserHandler.login()

Reply:
A little convoluted but it seems fine, Maybe add a comment referencing UserHandler.h where the translation from LOGIN to CONNECT is made. Also this frame will contain a port header which isn't necessary (but also shouldn't cause any problems)

Reply:
no problem with the added header, it's just more convenient. i think no comment is needed, as it is obvious